backend/ops_api/ops/resources/change_requests.py

```diff
@@ -0,0 +1,114 @@
+import copy
+from datetime import datetime
+
+import marshmallow_dataclass as mmdc
+from flask import Response, current_app, request
+from flask_jwt_extended import verify_jwt_in_request
+from sqlalchemy import select
+from typing_extensions import override
+
+from models import BudgetLineItem, BudgetLineItemChangeRequest, ChangeRequest, ChangeRequestStatus
+from ops_api.ops.auth.auth_enum import Permission, PermissionType
+from ops_api.ops.auth.decorators import is_authorized
+from ops_api.ops.base_views import BaseListAPI, handle_api_error
+from ops_api.ops.resources import budget_line_items
+from ops_api.ops.resources.budget_line_items import validate_and_prepare_change_data
+from ops_api.ops.schemas.budget_line_items import PATCHRequestBody
+from ops_api.ops.utils.response import make_response_with_headers
+from ops_api.ops.utils.user import get_user_from_token
+
+
+def review_change_request(
+    change_request_id: int, status_after_review: ChangeRequestStatus, reviewed_by_user_id: int
+) -> ChangeRequest:
+    session = current_app.db_session
+    change_request = session.get(ChangeRequest, change_request_id)
+    change_request.reviewed_by_id = reviewed_by_user_id
+    change_request.reviewed_on = datetime.now()
+    change_request.status = status_after_review
+
+    # If approved, then apply the changes
+    if status_after_review == ChangeRequestStatus.APPROVED:
+        if isinstance(change_request, BudgetLineItemChangeRequest):
+            budget_line_item = session.get(BudgetLineItem, change_request.budget_line_item_id)
+            # need to copy to avoid changing the original data in the ChangeRequest and triggering an update
+            data = copy.deepcopy(change_request.requested_changes)
+            schema = mmdc.class_schema(PATCHRequestBody)()
+            schema.context["id"] = change_request.budget_line_item_id
+            schema.context["method"] = "PATCH"
+
+            change_data, changing_from_data = validate_and_prepare_change_data(
+                data,
+                budget_line_item,
+                schema,
+                ["id", "status", "agreement_id"],
+                partial=False,
+            )
+
+            budget_line_items.update_data(budget_line_item, change_data)
+            session.add(budget_line_item)
+
+    session.add(change_request)
+    session.commit()
+    return change_request
+
+
+# TODO: Implement the queries needed for the For Approvals page, for now it's just a placeholder
+class ChangeRequestListAPI(BaseListAPI):
+    def __init__(self, model: ChangeRequest = ChangeRequest):
+        super().__init__(model)
+
+    @override
+    @is_authorized(PermissionType.GET, Permission.CHANGE_REQUEST)
+    @handle_api_error
+    def get(self) -> Response:
+        print("~~~GET ChangeRequestListAPI~~~")
```

Comment:
please remove print statements and commented out code

Reply:
done